docs/content/InputField.mdx

```diff
@@ -0,0 +1,192 @@
+</InputField>
+```
+
+### Using a TextInputWithTokens input
+
+```javascript live noinline
+const TextInputWithTokensExample = () => {
+  const [tokens, setTokens] = React.useState([
+    {text: 'zero', id: 0},
+    {text: 'one', id: 1},
+    {text: 'two', id: 2}
+  ])
+  const onTokenRemove = tokenId => {
+    setTokens(tokens.filter(token => token.id !== tokenId))
+  }
+
+  return (
+    <InputField>
+      <InputField.Label>Name</InputField.Label>
+      <TextInputWithTokens onTokenRemove={onTokenRemove} tokens={tokens} />
+    </InputField>
+  )
+}
+
+render(TextInputWithTokensExample)
+```
+
+### With a visually hidden label
+
+```jsx live
+<InputField>
+  <InputField.Label visuallyHidden>Name</InputField.Label>
+  <TextInput />
+</InputField>
+```
+
+### With a caption
+
+```jsx live
+<InputField>
+  <InputField.Label>Name</InputField.Label>
+  <TextInput />
+  <InputField.Caption>Hint: your first name</InputField.Caption>
+</InputField>
+```
+
+### With validation
+
+```javascript live noinline
+const ValidationExample = () => {
+  const [value, setValue] = React.useState('mona lisa')
+  const [validationResult, setValidationResult] = React.useState()
+  const doesValueContainSpaces = inputValue => /\s/g.test(inputValue)
+  const handleInputChange = e => {
+    setValue(e.currentTarget.value)
+  }
+
+  React.useEffect(() => {
+    if (doesValueContainSpaces(value)) {
+      setValidationResult('noSpaces')
+    } else if (value) {
+      setValidationResult('validName')
+    }
+  }, [value])
+
+  return (
+    <InputField
+      validationMap={{
+        noSpaces: 'error',
+        validName: 'success'
+      }}
+      validationResult={validationResult}
+    >
+      <InputField.Label>GitHub handle</InputField.Label>
+      <TextInput as={TextInput} block value={value} onChange={handleInputChange} />
+      <InputField.Validation validationKey="noSpaces">GitHub handles cannot contain spaces</InputField.Validation>
+      <InputField.Validation validationKey="validName">Valid name</InputField.Validation>
+      <InputField.Caption>With or without "@". For example "monalisa" or "@monalisa"</InputField.Caption>
+    </InputField>
+  )
+}
+
+render(ValidationExample)
+```
+
+## Props
+
+### InputField
+
+| Name             | Type                                                                                                                  |     Default      | Description                                                                                                                                                                                                          |
+| :--------------- | :-------------------------------------------------------------------------------------------------------------------- | :--------------: | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
```

Comment:
Are you planning to switch these to tables?

Reply:
Yes. I wrote this before `PropTable` was merged